provisioning/device/samples/ProvisioningDeviceClientX509/readme.md

```diff
@@ -0,0 +1,53 @@
+# Provisioning Device Client Sample - TPM Attestation
+
+## Overview
+
+This is a quick tutorial with the steps to register a device in the Microsoft Azure IoT Hub Device Provisioning Service using X509 certificates locally generated.
+
+## How to run the sample
+
+Ensure that all prerequisite steps presented in [samples](..\) have been performed.
+
+The sample code is set up to use X509 certificates stored within a password-protected PKCS12 formatted file (certificate.pfx). To generate a self-signed certificate run the following command:
+
+`powershell .\GenerateTestCertificate.ps1`
+
+The script will prompt for a PFX password. The same password must be used when running the sample.
+
+In your Device Provisioning Service go to "Manage enrollments" and select "Individual Enrollments".
+Select "Add" then fill in the following:
+Mechanism: X.509
+Certificate: Select the public key 'certificate.cer' file.
+DeviceID: iothubx509device1 (must match the CN part of the certificate Subject)
+
+To run the sample, in command prompt enter:
+`dotnet run <IDScope>`
+
+replacing `IDScope` with the value found within the Device Provisioning Service Overview tab. E.g. `dotnet run 0ne1234ABCD`
+
+Continue by following the instructions presented by the sample.
+
+## Using your certificates
+
+The SDK requires an [X509Certificate2](https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509certificate2(v=vs.110).aspx) object with private key ([HasPrivateKey](https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509certificate2.hasprivatekey(v=vs.110).aspx)==true) and, optionally, the certificate chain within an [X509Certificate2Colection](https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509certificate2collection(v=vs.110).aspx) object.
+
+This can be achieved by changing the following line:
+
+```C# 
+    using (var security = new SecurityClientX509(certificate))
+```
+
+to 
+
+```C# 
+    var myCertificate = new X509Certificate2("myCertificate.pfx", "mypassword");
+    var myChain = new X509Certificate2Collection();
+    myChain.Import("myChain.p7b");
+    using (var security = new SecurityClientX509(myCertificate, myChain)) 
+```
+
+A tool for creating _test_ certificates is available at https://github.com/Azure/azure-iot-sdk-c/blob/master/tools/CACertificates/CACertificateOverview.md
+
+If a Windows compatible Hardware Security Module is used, the certificate must be obtained by opening it from the Certificate Store using [X509Store](https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509store(v=vs.110).aspx).
```

Comment:
I just want to say thank you for linking to other MSDN articles.

Reply:
👍 Please let me know if we should link to other docs. (E.g. if you have a permanent link to your blog-posts.)